### ISSUE TITLE
pygame.error: Unable to open mouse

### ISSUE THREAD

**Issue by Nico4xo** [FdeAlexa/PeppyMeter_and_moOde#7]
Hi

Thx for Peppymeter !

I installed it by precisely following the tutorial on a raspberry pi 3 B with Moode 8.3.9 64 bits

1. I can't find "meter.size = xxxx" in config.txt I only see "meter = random"
Is this the right setting to change?

2. When starting peppymeter it gives me this:

Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "/home/moode/PeppyMeter/peppymeter.py", line 282, in <module>
    pm.init_display()
  File "/home/moode/PeppyMeter/peppymeter.py", line 143, in init_display
    pygame.display.init()
pygame.error: Unable to open mouse

I need a little help to resolve this, I'm not the sharpest knife in the drawer

Nico

**Comment by FdeAlexa** [comment]
Hi Nico,
the installation procedure illustrated here requires the default user to be "pi": in fact the problem you see is caused by the fact that "peppymeter" does not recognize the line "mouse.device = /dev/input/event0".
You could try modifying the config.txt by replacing "pi" with the name of your "default user name". In other words, try the following:

nano /home/USERNAME/PeppyMeter/config.txt

going to replace the content of only the lines below and leaving everything else unchanged:

"meter.size = large"
"framebuffer.device = /dev/fb0"
"mouse.device = /dev/input/event0"
"double.buffer = False"
"pipe.name = /var/tmp/peppyfifo"

Then save the file (Ctrl "o" and Enter) and exit (Ctrl "x").
Please let us know if you solve it.
Best regards,
Francesco

**Comment by Nico4xo** [comment]
Thank you Francesco

I reinstalled Moode with pi as user.

I still have the same problems:

No presence of the meter.size parameter in the config.txt file

> [current]
> 
> meter = random
> random.meter.interval = 20
> base.folder =
> meter.folder = 480x320
> screen.width =
> screen.height =
> exit.on.touch = False
> stop.display.on.touch = False
> output.display = True
> output.serial = False
> output.i2c = False
> output.pwm = False
> output.http = False
> use.logging = False
> use.cache = True
> frame.rate = 30
> 
> [sdl.env]
> framebuffer.device = /dev/fb0
> mouse.device = /dev/input/event0
> mouse.driver = TSLIB
> mouse.enabled = True
> video.driver = dummy
> video.display = :0
> double.buffer = False
> no.frame = False
> 
> [serial.interface]
> device.name = /dev/serial0
> baud.rate = 9600
> include.time = False
> update.period = 0.1
> 
> [i2c.interface]
> port = 1
> left.channel.address = 0x21
> right.channel.address = 0x20


> output.size = 10
> update.period = 0.1
> 
> [pwm.interface]
> frequency = 500
> gpio.pin.left = 24
> gpio.pin.right = 25
> update.period = 0.1
> 
> [http.interface]
> target.url = http://localhost:8000/vumeter
> update.period = 0.033
> 
> [web.server]
> http.port = 8001
> 
> [data.source]
> type = pipe
> polling.interval = 0.033
> pipe.name = /var/tmp/peppyfifo
> volume.constant = 80.0
> volume.min = 0.0
> volume.max = 100.0
> volume.max.in.pipe = 100.0
> step = 6
> mono.algorithm = average
> stereo.algorithm = new
> smooth.buffer.size =  4

And sudo python3 peppymeter.py gives:

> pygame 1.9.6
> Hello from the pygame community. https://www.pygame.org/contribute.html
> Traceback (most recent call last):
>   File "/home/pi/PeppyMeter/peppymeter.py", line 282, in <module>
>     pm.init_display()
>   File "/home/pi/PeppyMeter/peppymeter.py", line 143, in init_display
>     pygame.display.init()
> pygame.error: Unable to open mouse

Note that I had a lot of warnings during the installation of PeppyAlsa

> aclocal && libtoolize
> autoconf && automake --add-missing
> ./configure && make

And that step

>git clone https://github.com/project-owner/PeppyMeter.git

Did not create the PeppyMeter directory in the pi directory and I had to redo it after installing Python

I am desperate !

Nico

**Comment by FdeAlexa** [comment]
Hello Nico,
I have just tried a new installation from a scratch 8.3.9.  rds,
Frahere is not longer the parameter "meter.size": I have already corrected the instructions. The parameter "meter.folder" has to be setted at "800x480" for the display standard rpi 7".
The error you reported is related to the input device: it seems that you are not using the mentioned display screen: would you please  specify what is your hardware configuration?
Regards,
Francesco

**Comment by Nico4xo** [comment]
Hi Francesco

Sorry for the time taken to respond

moOde 8.3.9 (64-bit Bullseye)
Rpi 3B 1Go
IQaudio DAC+

I realized that I was trying to display peppymeter on a 1920*1080 tv and that it was not the best thing

So in the meantime I bought a 3.5" waveshare lcd to put on the gpio

But... I already can't display Moode on it... It remains stuck on the startup screen and moode is displayed via hdmi in 480*320 on my 1920*1080 screen

It's difficult when you don't know much about it...

I guess it's a configuration problem.

config.txt :

> [cm4]
> otg_mode=1
> 
> [all]
> disable_overscan=1
> display_rotate=0
> disable_splash=1
> dtoverlay=disable-wifi
> dtoverlay=disable-bt
> dtoverlay=iqaudio-dacplus
> dtoverlay=waveshare35a
> dtparam=audio=off
> dtparam=i2s=on
> dtparam=i2c_arm=on
> dtparam=spi=on
> enable_uart=1
> hdmi_blanking=1
> hdmi_cvt 480 320 60 6 0 0 0
> hdmi_drive=2
> hdmi_force_hotplug=1
> hdmi_force_edid_audio=1
> hdmi_force_hotplug=1
> hdmi_group=2
> hdmi_mode=1
> hdmi_mode=87
> max_usb_current=1

I followed the following tutorial to install it : 

https://www.waveshare.com/wiki/3.5inch_RPi_LCD_(A)_Manual_Configuration#Set_Auto-start

**Comment by FdeAlexa** [comment]
Hi Nico,
let's try to solve the issue related to PeppyMeter first: with an HDMI display only;  I tried a configuration similar to yours: rpi 3, old Samsung TV (1920 x 1080) attached to HDMI. If you connect a USB mouse to the Raspberry the error disappared and with some modification to the PeppyMeter/config.txt it display the meter.
The modifications I did are:
meter.folder = 1280x400                     (because it is enough big the display resolution)
screen.width = 1920                            (it is the real display width)
screen.height = 1080                           (it is the real display heigh)  (you can reach them with the "tvservice -v 2 -s" command)
framebuffer.device = /dev/fb0
mouse.device = /dev/input/event0
pipe.name = /var/tmp/peppyfifo
I also used a fixed meter (meter = emerald) only for test purposes, but you can leave the default or another meter you prefere.

Please find below the full config.txt

[current]
meter = emerald
random.meter.interval = 20
base.folder =
meter.folder = 1280x400
screen.width = 1920
screen.height = 1080
exit.on.touch = False
stop.display.on.touch = False
output.display = True
output.serial = False
output.i2c = False
output.pwm = False
output.http = False
use.logging = False
use.cache = True
frame.rate = 30

[sdl.env]
framebuffer.device = /dev/fb0
mouse.device = /dev/input/event0
mouse.driver = TSLIB
mouse.enabled = True
video.driver = dummy
video.display = :0
double.buffer = False
no.frame = False

[serial.interface]
device.name = /dev/serial0
baud.rate = 9600
include.time = False
update.period = 0.1

[i2c.interface]
port = 1
left.channel.address = 0x21
right.channel.address = 0x20
output.size = 10
update.period = 0.1

[pwm.interface]
frequency = 500
gpio.pin.left = 24
gpio.pin.right = 25
update.period = 0.1

[http.interface]
target.url = http://localhost:8000/vumeter
update.period = 0.033

[web.server]
http.port = 8001

[data.source]
type = pipe
polling.interval = 0.033
pipe.name = /var/tmp/peppyfifo
volume.constant = 80.0
volume.min = 0.0
volume.max = 100.0
volume.max.in.pipe = 100.0
step = 6
mono.algorithm = average
stereo.algorithm = new
smooth.buffer.size = 4

Nico, let me know if you solve the problem.
Best regards,
Francesco

**Comment by Nico4xo** [comment]
Thank you very much Francesco

I applied your changes and successfully displayed.

Then I imported a file with skins in 1920x1080 and now I have everything perfectly!

Furthermore, I returned the small 480x320 screen, I didn't like it!

Thanks to you and now that I understand better how it works I will be able to move forward with my project.

I want to get a streaming box with a screen and I absolutely wanted it to be with Moode + Peppymeter. I will be able to concentrate on the power supply/rpi/dac part now.

Thank you again for your help and patience.

Nico
